grapher/core/Grapher.tsx

```diff
@@ -1090,11 +1090,16 @@ export class Grapher
     }
 
     @computed get hasTimeline() {
-        if (this.isStackedBar || this.isStackedArea || this.isDiscreteBar)
-            return false
         if (this.isOnOverlay) return false
         if (this.hideTimeline) return false
-        if (this.isOnMapTab && this.map.hideTimeline) return false
+
+        if (this.isOnMapTab) {
+            if (this.map.hideTimeline) return false
+        } else {
+            if (this.isStackedBar || this.isStackedArea || this.isDiscreteBar)
+                return false
+        }
+
```

Comment:
The "if map" statement should be before the `if (this.hideTimeline)` check to match the behaviour on live?

Otherwise hiding the timeline on a chart would hide it for a map too

Reply:
Ah but that change alone would lead to the Table tab not having a timeline on stacked bar/area and discrete bar, so might need another if statement

Reply:
Good catch! I've now re-written that entire method into a switch/case statement, which I find to be more expressive in this case. It should also fix the two cases you outlined here, and makes most conditions on a tab-dependent basis so we're able to reason about it more easily.

Let me know if you like it!

Reply:
🙌 This is much better! 

I'd add a `: boolean` return type to `hasTimeline()` which would give a type error if some tabs are not handled in the `switch`. Or alternatively there could be a `default: return false` instead of the download & sources tabs?

Also, can get rid of the `get isOnOverlay()` now.

Reply:
Ah, good point about the `boolean` type.
I intentionally split `download` and `sources` up and didn't introduce a `default`; this way we will have to re-visit if we ever add/remove new tab possibilities and cannot forget about it.

I also got rid of `isOnOverlay` now. It sounded like a method that could be nice-to-have in the future, but since it is private anyway let's get rid of it now, and we can just recreate it whenever we should need it 👍